### PR TITLE
Code Health | Enforcing Ordinal for  StringComparison

### PR DESCRIPTION
**Pull request by JRahnama**
@Wraith2: based on the other conversation we had on PR #2065, what do you think on this change request?

### CODE REVIEW

**Comment by ErikEJ** [comment]
I was just going to ask if there were other potential  relevant changes - LGTM

**Comment by Wraith2** [comment]
Looks like a good idea to me.

File: .editorconfig [hunk 1]
```diff
@@ -143,6 +143,9 @@ dotnet_diagnostic.CA2100.severity = silent
 # CA1416: Validate platform compatibility
 dotnet_diagnostic.CA1416.severity = silent
 
+# CA1310: Specify StringComparison for correctness
+dotnet_diagnostic.CA1310.severity = error
```
Comment (David-Engel) [inline .editorconfig:147]: As you can see from the build results, we should consider putting this in a .editorconfig file in various subfolders to only apply it to libraries that require it.

YES: MDS, MSS, MDS AKV Provider
MAYBE: Test utility libraries
NEEDS EVAL: MDS Test libraries (I'm guessing a lot of places in test code should be specifying the option, but not sure if all require it, so we might not want to impose this rule on tests.)

**Comment by David-Engel** [comment]
To add background for anyone investigating this in the future: In a library like MDS, almost all string comparison operations being done inside the library should not use culture comparison rules (they should use Ordinal). For example, if we are parsing a connection string, we are looking for specific characters, not culture-specific representations of them. For example, if we are looking for a comma (also a thousands separator in some languages), we only want to find a comma, not a period (also a thousands separator in some languages). Default string comparison operations in .NET are culture specific. To be safe, we should always be explicit with which option we want to use inside MDS.

File: .editorconfig (hunk 1 above)
Comment (JRahnama) [reply .editorconfig:147]: that is what I was looking for, but nothing on the .editorconfig subfolders helped me. so I went through all the tests and noticed there are almost 17 classes that needs to be converted to ordinal comparison. There were couple of places that ignore case were required. I have changed them all. If you still think that I should go with the editorconfig setup. I will revert these changes and will go with that pattern.

**Review by David-Engel — APPROVED** [review]

**Review by Kaur-Parminder — APPROVED** [review]

**Pull request merged**